### DISCUSSION TITLE
is Buster ok ?

### DISCUSSION THREAD

**Discussion by laorka** [1000001101000/Debian_on_Buffalo#96] [Q&A] [answered]
Hi, thanks a lot for your work, it's way more faster with Debian compared to the custom build from Buffalo...
Is updating to Debian 10 ok ? (running on a LS220DR)

**Comment by 1000001101000** [comment, marked as answer]
Good to hear from you!

Are you currently running Stretch?

Buster works well on that device though I've only really tested doing fresh installs rather than using the dist-upgrade process. 

If you do decide to upgrade rather than a fresh install you need to be careful to make sure that your device-tree always matches the kernel you have installed otherwise the device may fail to boot. 

Once you've installed the new 4.19 kernel for buster (before you reboot), copy the new DTB into /etc/flash-kernel/dtbs/ and then run `flash-kernel` to install it. You can get the Buster dtb here:
https://github.com/1000001101000/Debian_on_Buffalo/tree/master/Buster/device_trees

**Reply by laorka** [reply]
Yes I was running Stretch, I upgraded to Buster and flashed kernel

It's booting without issue and the system seems to be stable

**Reply by 1000001101000** [reply]
Excellent!

I went through the process for the first time on my desktop a few months ago which was mostly painless. It did result in some packages that broke and had to be removed which it took me a while to notice.
